lib/uplink/bucket_attrs_test.go

```diff
@@ -57,6 +58,45 @@ func simpleEncryptionAccess(encKey string) (access EncryptionAccess) {
 	return access
 }
 
+func createPartnerID() (partnerID string, err error) {
+	id, err := uuid.New()
+	if err != nil {
+		return partnerID, err
+	}
+	return id.String(), err
+}
+
+// check that partner bucket attributes are stored and retrieved correctly.
+func TestPartnerBucketAttrs(t *testing.T) {
+	var (
+		access     = simpleEncryptionAccess("voxmachina")
+		bucketName = "mightynein"
+	)
+
+	testPlanetWithLibUplink(t, testConfig{},
+		func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet, proj *Project) {
+			_, err := proj.CreateBucket(ctx, bucketName, nil)
+			require.NoError(t, err)
+
+			partnerID, err := createPartnerID()
+			require.NoError(t, err)
+
+			// partner ID set
+			proj.uplinkCfg.Volatile.PartnerID = partnerID
+			got, err := proj.OpenBucket(ctx, bucketName, &access)
+			require.NoError(t, err)
+
+			got, err = proj.OpenBucket(ctx, bucketName, &access)
+			require.NoError(t, err)
```

Comment:
The second openBucket actually returns from a different part of the code which is when a bucket is already attributed , just returns nil with a debug info logged on the satellite side, which it was from the previous OpenBucket(), but there isn't anyway to test it.  I removed it just to avoid confusion.